sdk/identity/identity/src/errors.ts

```diff
@@ -75,8 +75,9 @@ export const CredentialUnavailableErrorName = "CredentialUnavailableError";
  * an error that should halt the chain, it's caught and the chain continues
  */
 export class CredentialUnavailableError extends Error {
-  constructor(message?: string) {
-    super(message);
+  constructor(message?: string, options?: { cause?: unknown }) {
+    // @ts-expect-error - TypeScript does not recognize this until we use ES2022 as the target; however, all our major runtimes do support the `cause` property
```

Comment:
Could we update the type instead of suppressing the type error?

```ts
interface Error {
  name: string;
  message: string;
  cause?: any;
  stack?: string;
}

interface ErrorConstructor {
  new(message?: string, options?: { cause: any }): Error;
  (message?: string): Error;
  readonly prototype: Error;
}

declare var Error: ErrorConstructor
```

Reply:
We can! that does work (although I have to suppress one of two rules: either [eslint/novar](https://eslint.org/docs/latest/rules/no-var) or [typescript-eslint/no-redeclare](https://typescript-eslint.io/rules/no-redeclare/) depending on whether we use `var` or `const` ) 

The benefit of using `ts-expect-error` is that once we update our target to ES2022 this will automatically be caught by the compiler (it'll complain if ts-expect-error is set but there's no error) which will help us remember to remove this (and use ErrorOptions instead of having our own).

I'll play around with both though and see what works better - thanks for the suggestion!!